### ISSUE TITLE
[Feature Request] Compare current working tree against a commit

### ISSUE THREAD

**Issue by Santjo** [ForkIssues/TrackerWin#722]
Hi, 
I am not sure but it seems that it is not possible yet to compare the current working directory (which is not commited) against a specific commit.  

br santjo

**Comment by DanPristupov** [comment]
Fork doesn't have this feature and I don't know a reliable way to compare a commit and working directory.

P.S. please fix the title

**Comment by Santjo** [comment]
hi Dan, 
how does it TortoiseGit or SourceTree ? New added files are not diaplyed but differnces between existing files.  

I mean the call:
`git diff [<options>] [<commit>] [--] [<path>]
`e.g. 
`git diff --name-only 84533e54e4106ac077ee187ae4bb34656c6cf502 .`

**Comment by DanPristupov** [comment]
> New added files are not diaplyed but differnces between existing files.

Yes, this is the problem. New files must be included, otherwise it would be really confusing.

**Comment by ChuckM2** [comment]
I would also really like this feature.  I use it all the time in TortoiseGit.

**Comment by pborsutzki** [comment]
Nice improvement! Unfortunately, opening a file in the `Local Changes` view using the `External Diff` function does not use the file from the repository for comparison, but some new copy `temp/file~old.ext` of it. This defeats the purpose of this feature for me, as I am unable to change the files in the repo based on the comparison then (like I could with TortoiseGit). Could you change this behavior?

**Comment by DanPristupov** [comment]
@pborsutzki I confirm, the external diff feature doesn't work for the working directory comparison.

**Comment by ChuckM2** [comment]
The `External Diff` is also not working correctly for me.  When I compare to local changes, the UI diff in fork looks correct.  But the external diff opens the wrong files for the compare and the two appear identical.  For example, if I compare SHA 12345 to local, the "old" file name is `foo~old.h` and the "new" file name is `foo~12345.h`.  The "old" file should be `foo~12345.h` and the "new" file should be the local file in the workspace.  Can this please be fixed?

**Comment by GioCC** [comment]
I would also like to add that - beside it not being working - it's not at all clear how, or whether, it's possible to compare the working directory with any given commit.
The context menu entry in the "local changes" view seems to only compare to the latest commit.
Could you please reconsider looking at this feature?

It also works well in _GitExtension_ (where an entry for the Index is also provided):
![Screenshot_1](https://github.com/user-attachments/assets/e2fc5b98-31f5-40c2-bd3e-25f1546352e1)

**Comment by DanPristupov** [comment]
> It's not at all clear how, or whether, it's possible to compare the working directory with any given commit.

Right click on a commit and select `Compare to Local Changes`

**Comment by Kamik423** [comment]
It is great that there is actually a way to do this, I had not discovered it.

I am still of the opinion that showing the working copy in the tree makes this feature more discoverable and intuitive, so I would still appreciate having it.
